Library/Homebrew/cmd/reinstall.rb

```diff
@@ -2,6 +2,14 @@
 
 module Homebrew
   def reinstall
+    if !MacOS.can_build?
```

Comment:
Very similar logic is being repeated in a few places; maybe this could be moved into a class method on `FormulaInstaller`?